src/vmm/src/device_manager/persist.rs

```diff
@@ -429,13 +433,15 @@ mod tests {
             };
             insert_vsock_device(&mut vmm, &mut cmdline, &mut event_manager, vsock_config);
             // This will be used by the restored device and will cleanup the UDS when test ends.
-            _tmp_sock_file = orig_tmp_sock_file.clone();
+            _tmp_sock_file = orig_tmp_sock_file;
```

Comment:
Removing/unlinking a file/socket/pipe/whatever (everything is a _file_ in unix) from the filesystem (`fs::remove()` in Rust), does not impair any functionality on it in processes which had previously opened it and still have an open `file descriptor` for it.

What we're interested here is unlinking the socket from the filesystem - which translates in freeing up the FS entry for `/path/to/file` - so that the block/net/vsock device can use that path to create their file/tap/uds. The path/name needs to be free.

We just merged #1928 which simplifies the logic here.

@timvisee Try rebasing this PR on top of master and it should work.

Reply:
I've just rebased. I also dropped my last commit because it looks like it's obsolete. We'll see what CI reports.

Reply:
```rust
-        _tmp_sock_file = orig_tmp_sock_file.clone();
+        _tmp_sock_file = orig_tmp_sock_file;
```
This was the breaking change in your PR.

The `clone()` there was seen as redundant by `clippy`, but it had its purpose:

to have **two** objects that cleanup their internal path on `drop()`. The first object would clean up at the end of the code-block, and the second at the end of the test.

Without the `clone()`, the internal path was not getting cleaned up at the end of the code-block so the code coming after which did the device restore failed because device restore failed with `AddrInUse` error.

Reply:
@acatangiu Could you locate the line this is on, so I can fix it?

Reply:
@timvisee that breaking change it is not relevant anymore. It was fixed by a previous merged PR. It seems that there is a coverage issue at the moment, although the changes does not bring new code. I think you can adjust the coverage, as per report. You can find it in `tests/build/test_coverage.py`.

Reply:
> that breaking change it is not relevant anymore

Missed that this was a comment on a earlier review. Whoops!

I've changed the required coverage percentage now.